File: imageSearch.py
def imageSearch(images, term):
    lenOfImages = len(images)
    imagesCasefold = [image.casefold() for image in images]
    termCasefold = term.casefold()
    imagesToRemove = []

    # determine if term is in images using casefold to make search case insensitive
    for image in imagesCasefold:
        if term not in image:
            # indexes in images and imagesCasefold should match
            # so I can use the the current index to grab the exact image name that I'm going to remove next
            imageIndex = imagesCasefold.index(image)
            imagesToRemove.append(images[imageIndex])

    # removes the images that don't match the term
    for image in imagesToRemove:
        imageIndex = images.index(image)
        del images[imageIndex]

    return images

print(imageSearch(["Sunset.jpg", "Beach.png", "sunflower.jpeg"], "sun"))

File: imageSearch.js
function imageSearch(images, term) {
  var lenOfImages = images.length;
  var imagesToRemove = [];

//   First for loop checks for matches and saves the ones that don't match to a separate array

  for (let i = 0; i < lenOfImages ; i++){
    let currentImage = images[i];
// the regular expresions allows for more complex search methods and lets me be case insenstive
    let regExSearch = new RegExp(term, "gi")
    var matchArr = currentImage.match(regExSearch)
    // match returns null if it doesn't match so I push it to the separate array
    if (matchArr == null) {
      imagesToRemove.push(currentImage)
    }
  }

  var lenofRemoval = imagesToRemove.length;

//   Second for loop finds the index of the image to remove and removes it
// Not doing it this way makes the for loop above break because the length of the images array would change if 
// I took it out right then
// There's probably a way to condense this but I'm not sure
  for(let i=0; i < lenofRemoval; i++){
    var itemToRemove = imagesToRemove[i];
    var index = images.indexOf(itemToRemove);
    if (index !== -1) {
        images.splice(index, 1);
    }
  }

  return images;
}

imageSearch(["dog.png", "cat.jpg", "parrot.jpeg"], "dog")